src/Controller/TaskController.php

```diff
@@ -48,23 +52,34 @@ public function update(
     #[Route('tasks', name: 'get_tasks', methods: 'GET')]
     public function list(TaskCalendarRepository $taskCalendarRepository, Request $request): JsonResponse
     {
-        $page = (int) $request->query->get('page', 1);
-        $per_page = (int) $request->query->get('per_page', 10);
-        $sort = $request->query->get('sort', null);
+        try {
+            $page = (int) $request->query->get('page', 1);
+            $per_page = (int) $request->query->get('per_page', 10);
+            $sort = $request->query->get('sort', null);
+            $filter = $request->query->get('filter', null);
 
-        $tasks = $taskCalendarRepository->getAllPaginated($page, $per_page, $sort);
+            $tasks = $taskCalendarRepository->getPaginatedWithFilterAndSort($page, $per_page, $sort, $filter);
 
-        return $this->json(
-            $tasks,
-            Response::HTTP_OK,
-            [],
-            [
-                AbstractNormalizer::CIRCULAR_REFERENCE_HANDLER => function ($object) {
-                    return $object->getId();
-                },
-                AbstractNormalizer::CIRCULAR_REFERENCE_LIMIT => 1,
-            ]
-        );
+            return $this->json(
+                $tasks,
+                Response::HTTP_OK,
+                [],
+                [
+                    AbstractNormalizer::CIRCULAR_REFERENCE_HANDLER => function ($object) {
+                        return $object->getId();
+                    },
+                    AbstractNormalizer::CIRCULAR_REFERENCE_LIMIT => 1,
+                ]
+            );
+        } catch (QueryException $e) {
+            return $this->json([], Response::HTTP_BAD_REQUEST);
+        } catch (InvalidOperatorException $e) {
+            return $this->json([], Response::HTTP_BAD_REQUEST);
+        } catch (InvalidFilterException $e) {
+            return $this->json([], Response::HTTP_BAD_REQUEST);
+        } catch (InvalidFormatException $e) {
+            return $this->json([], Response::HTTP_BAD_REQUEST);
+        }
```

Comment:
Staramy się tak nie robić. Jeśli potrzebujesz przechwycić w jakiś customowy sposób wyjątek to powinieneś zrobić customowego error handlera. W Symfony 6.3 jest możliwość tworzenia customowych wyjątków na bazie HTTPException, dzięki temu w samej klasie wyjątku możesz ustawić kod HTTP zwracany w odpowiedzi: https://symfony.com/blog/new-in-symfony-6-3-http-exception-attributes